### ISSUE TITLE
Postcode / Zipcode in checkout form already validated on page load 

### ISSUE THREAD

**Issue by dverkade** [magento/magento2#18630]
### Preconditions
1. Magento 2.2.6

### Related issues

There are several bugs about this, but they are all closed and it's mentioned that this has been resolved in Magento 2.2.X, but unfortunately it's not in the latest version, so opening a new bug report for this issue. 

1. magento/magento2#7724
2. magento/magento2#8115

### Steps to reproduce (*)
1. Add products to cart
2. Checkout as a guest
3. The `tax/defaults/postcode` config value in `core_config_data` is NULL

### Expected result (*)
1. Postcode / Zipcode field should be empty and should not be validated 
2. No error message should be shown that the field is required

### Actual result (*)
1. Postcode / Zipcode field has error message that the field is required

![image](https://user-images.githubusercontent.com/1058480/46969519-9c515000-d0b6-11e8-9ddc-245130640302.png)

**Comment by magento-engcom-team** [comment]
Hi @dverkade. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@dverkade do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
HI @dverkade thank you for your report, i'm confusing with your issue, why zip code must be not required ? after you chose tax/default/postcode ? because this field is required -> Settings can be found in: Admin Panel > Stores > Settings > Configuration > General > General > Country options > Zip/Postal Code is Optional for .
So if you chose all you dont have this error validation. And why taxes  can remove validation from aip postal code ?

**Comment by ghost** [comment]
@dverkade Ok, i'm understand that if page load, then this validate the zip code filed, but in my case  i have no validation, see attachments.
![zip](https://user-images.githubusercontent.com/40756439/47645981-30252080-db7b-11e8-808e-56d8096c8278.gif)

**Comment by dverkade** [comment]
@engcom-backlog-nazar if I look at the screenshots you are using Magento 2.3 development. As noted in the issue description we have this issue with Magento 2.2.6.

**Comment by ghost** [comment]
@dverkade ok, but you created the PR https://github.com/magento/magento2/pull/18633 to 2.3-develop so this issue must be in 2.3 ?

**Comment by ghost** [comment]
@dverkade the same on 2.2-develop and 2.2.6 instance, i'm do exactly what you mentioned on steps to reproduce.

**Comment by ghost** [comment]
@dverkade Can you reproduce this on vanilla instance ?

**Comment by ghost** [comment]
@dverkade, we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by ghost** [comment]
I'm able to reproduced this issue by following steps on 2.2-develop and 2.3-develop: 
1. Use system value to : Stores > Settings > Configuration > General > General > Country options > Zip/Postal Code is Optional for
2. Add products to cart as guest
3. Proceed to checkout

**Comment by magento-engcom-team** [comment]
@engcom-backlog-andrii Thank you for verifying the issue.

Unfortunately, not enough information was provided to created internal ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@engcom-backlog-andrii Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96736`, `MAGETWO-96737` were created

**Comment by PathToLife** [comment]
Can also reproduce.. 

Still exists in 2.3.0. Just have to set a postcode in "Default Tax Destination Calculation"

Which is set here:
http://Base_url/admin/admin/system_config/edit/section/tax/
![postcode set here](https://i.imgur.com/LNSjoKT.png)

Use Xdebug in any active shipping module on calling RateRequest->getDestPostcode(). We observe that:
```php
Magento\Quote\Model\Quote\Address\RateRequest->$_data['dest_postcode'] = Default Tax Destination Calculation
```

**Comment by magento-engcom-team** [comment]
Hi @dverkade. Thank you for your report. 
The issue has been fixed in magento/magento2#18633 by @dverkade in 2.3-develop branch
Related commit(s):
 - [bf024c1c1f3b86a7c3dd362b78fb124f58f22b96](https://github.com/magento/magento2/commit/bf024c1c1f3b86a7c3dd362b78fb124f58f22b96)
 - [df90f67822ea0f1c5765b512c1d7613532a41a11](https://github.com/magento/magento2/commit/df90f67822ea0f1c5765b512c1d7613532a41a11)
 - [6257fe8a3938c59e207b51da30202e1f5a65eb76](https://github.com/magento/magento2/commit/6257fe8a3938c59e207b51da30202e1f5a65eb76)
 - [ac11b348a1f4327106474c7734b4fa65102d3ca8](https://github.com/magento/magento2/commit/ac11b348a1f4327106474c7734b4fa65102d3ca8)

The fix will be available with the upcoming 2.3.2 release.
